### PR TITLE
CR85 First potentially working version of get case by UPRN

### PR DESCRIPTION
**Pull request by PeterBochel**
# Motivation and Context
This pull request is for the implementation of the get case by UPRN endpoint.

# What has changed
Fake implementation replaced with code that makes a http call to the case service.

# How to test?
Run all unit tests..

# Links
See: https://collaborate2.ons.gov.uk/jira/browse/CR-85

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java [hunk 1]
```diff
@@ -35,8 +38,9 @@ public CaseDTO getCaseById(final UUID caseId, CaseRequestDTO requestParamsDTO) {
     CaseContainerDTO caseDetails = caseServiceClient.getCaseById(caseId, getCaseEvents);
 
     // Only return Household cases
-    if (!caseDetails.getCaseType().equals(CaseType.H.name())) {
-      throw new ResponseStatusException(HttpStatus.FORBIDDEN, "Case is a non-household case");
+    if (!caseIsHouseholdOrCommonual(caseDetails.getCaseType())) {
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java]: Communal

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplTest.java [hunk 2]
```diff
@@ -46,20 +51,20 @@ public void initMocks() {
...
   @Test
   public void testGetCaseByCaseId_withCaseDetails() throws Exception {
-    doTestGetCaseByCaseId(true);
+    doTestGetCaseByCaseId(CaseType.H, true);
   }
 
   @Test
   public void testGetCaseByCaseId_withNoCaseDetails() throws Exception {
-    doTestGetCaseByCaseId(false);
+    doTestGetCaseByCaseId(CaseType.C, false);
```
Comment (philwhiles) [inline src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplTest.java]: testGetCaseByCaseId_withCaseDetails and testGetCaseByCaseId_withNoCaseDetails use two different args when calling doTestGetCaseByCaseId - should they not vary the criteria more subtly or are more variations of this test required?

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java (hunk 1 above)
Comment (PeterBochel) [reply src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java]: Typo fixed.

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplTest.java (hunk 2 above)
Comment (PeterBochel) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplTest.java]: More permutations added. 

I was taking a shortcut, and slipping testing of the household/communal filtering into the events on/off tests.

**Comment by PeterBochel** [comment]
> Speling correktion please

**Pull request merged**